File: test/golden/java/src/main/java/com/example/testmod/TestmodMod.java
package com.example.testmod;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/*? if fabric {*/
import net.fabricmc.api.ModInitializer;
/*?}*/

/*? if neoforge {*/
/*import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.common.Mod;
*//*?}*/

/*? if neoforge {*/
/*@Mod(TestmodMod.MOD_ID)
public class TestmodMod {
    public TestmodMod(IEventBus modEventBus) {
        init();
    }
*//*?} elif fabric {*/
public class TestmodMod implements ModInitializer {
    @Override
    public void onInitialize() {
        init();
    }
/*?}*/

    public static final String MOD_ID = "testmod";
    public static final Logger LOGGER = LoggerFactory.getLogger(MOD_ID);

    public static void init() {
        LOGGER.info("Initializing Test Mod");
    }
}

File: test/golden/kotlin/src/main/kotlin/com/example/testmod/TestmodMod.kt
package com.example.testmod

import org.slf4j.LoggerFactory

/*? if fabric {*/
import net.fabricmc.api.ModInitializer
/*?}*/

/*? if neoforge {*/
/*import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.common.Mod
*//*?}*/

/*? if neoforge {*/
/*@Mod(TestmodMod.MOD_ID)
class TestmodMod(modEventBus: IEventBus) {
    init {
        Companion.init()
    }
*//*?} elif fabric {*/
class TestmodMod : ModInitializer {
    override fun onInitialize() {
        init()
    }
/*?}*/

    companion object {
        const val MOD_ID = "testmod"
        val LOGGER = LoggerFactory.getLogger(MOD_ID)

        fun init() {
            LOGGER.info("Initializing Test Mod")
        }
    }
}
